src/librustc_mir/build/expr/as_temp.rs

```diff
@@ -40,9 +40,9 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         let source_info = this.source_info(expr_span);
 
         if expr.temp_lifetime_was_shrunk && this.hir.needs_drop(expr_ty) {
-            this.hir.tcx().sess.span_err(
+            this.hir.tcx().sess.span_warn(
                 expr_span,
-                "this expression will be dropped at a different time - see #38603");
+                "this expression used to live longer due to #36082");
```

Comment:
we should make a proper tracking issue with an explanation of how to rewrite code to be unambiguous, but this seems good otherwise